web/src/app/settings/suppliers/hooks.ts

```diff
@@ -307,25 +197,66 @@ export const useHook = () => {
   ]
 
 
-  const onSubmit: SubmitErrorHandler<ISupplier> = (value) => {
-    if (btnName === "Save changes") {
-      console.log("handle update ", value)
+  const onSubmit: SubmitHandler<SupplierDto> = async (value) => {
+    setIsLoading(true)
+    try {
+      //prevent multiple submit one at a time
+      if (!isLoading) {
+        if (btnName === "Save changes") {
+          console.log("handle update ", value)
+          const payload: UpdateSupplierDto = {
+            supplier: value.supplier,
+            canvassAllowed: value.canvassAllowed,
+            address: value.address ?? "N/A",
+            contact: value.contact ?? "N/A",
+            email: value.email ?? "N/A",
+            nsid: value.nsid ?? "N/A",
+            proprietor: value.proprietor ?? "N/A",
+            tin: value.tin ?? "N/A"
+          }
+          await handleUpdate({ id: value.id, payload })
+          setSnackbarProps({ open: true, message: "Successfully update." })
```

Comment:
try check sa main component if default and `open` na property
```suggestion
          setSnackbarProps({  message: "Successfully update." })
```